internal/config/validate.go

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package config
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+type Validate func(*configFile) error
+
+// DiscardFRROnly returns an error if the current configFile contains
+// any options that are available only in the FRR implementation.
+func DiscardFRROnly(c *configFile) error {
+	for _, p := range c.Peers {
+		if p.BFDProfile != "" {
+			return fmt.Errorf("peer %s has bfd profile set on native bgp mode", p.Addr)
+		}
+		if p.KeepaliveTime != "" {
+			return fmt.Errorf("peer %s has keepalive set on native bgp mode", p.Addr)
+		}
+	}
+	if len(c.BFDProfiles) > 0 {
+		return errors.New("bfd profiles section set")
+	}
+	for _, p := range c.Pools {
+		for _, a := range p.BGPAdvertisements {
+			if a.AggregationLengthV6 != nil {
+				return fmt.Errorf("pool %s has aggregation lenghtv6 was set on native bgp mode", p.Name)
```

Comment:
typo in the error msg

Reply:
Good catch, I also reworked the errors so it's clear what field they refers to.